3p/recaptcha.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {IframeMessagingClient} from './iframe-messaging-client';
+import {dict} from '../src/utils/object';
+import {user} from '../src/log';
+import {writeScript} from './3p';
+
+/** @const {string} */
+const TAG = 'RECAPTCHA';
+
+/** {!IframeMessaginClient|null} **/
+let iframeMessagingClient = null;
+
+/**
+ * Get the recaptcha script.
+ *
+ * Use writeScript: Failed to execute 'write' on 'Document': It isn't possible
+ * to write into a document from an asynchronously-loaded external script unless
+ * it is explicitly opened.
+ *
+ * @param {!Window} global
+ * @param {string} scriptSource The source of the script, different for post and comment embeds.
+ * @param {function(*)} cb
+ */
+function getRecaptchaApiJs(global, scriptSource, cb) {
+  writeScript(global, scriptSource, function() {
+    cb(global.grecaptcha);
+  });
+}
+
+/**
+ * Function to handle executing actions using the grecaptcha Object,
+ * and sending the token back to the parent amp-recaptcha component
+ *
+ * @param {*} grecaptcha
+ * @param {string} siteKey
+ * @param {Object} data
+ */
+function actionTypeHandler(grecaptcha, siteKey, data) {
+  if (!iframeMessagingClient) {
+    user().error(TAG, 'IframeMessagingClient does not exist.');
+    return;
+  }
+
+  grecaptcha.execute(siteKey, {
+    action: data.action,
+  })./*OK*/then(function(token) {
```

Comment:
So the reason I haven't been using phat arrow functions is because I noticed the other `3p` scripts use the normal function callbacks. I think the reasoning is that the other 3p scripts may be from when we still supported IE11, which does not support the phat arrow ES6 syntax. And since I am assuming we don't have a phat arrow polyfill for the 3p scripts, I think we should stick with this style.

Reply:
Sounds good! Thanks for the detailed explanation.